data/transactions/logic/eval.go

```diff
@@ -612,26 +612,226 @@ func (cx *EvalContext) RunMode() RunMode {
 	return cx.runModeFlags
 }
 
-// StackType describes the type of a value on the operand stack
-type StackType byte
+// avmType describes the type of a value on the operand stack
+type avmType byte
 
 const (
-	// StackNone in an OpSpec shows that the op pops or yields nothing
-	StackNone StackType = iota
+	// avmNone in an OpSpec shows that the op pops or yields nothing
+	avmNone avmType = iota
+
+	// avmAny in an OpSpec shows that the op pops or yield any type
+	avmAny
 
-	// StackAny in an OpSpec shows that the op pops or yield any type
-	StackAny
+	// avmUint64 in an OpSpec shows that the op pops or yields a uint64
+	avmUint64
 
-	// StackUint64 in an OpSpec shows that the op pops or yields a uint64
-	StackUint64
+	// avmBytes in an OpSpec shows that the op pops or yields a []byte
+	avmBytes
+)
+
+func (at avmType) String() string {
+	switch at {
+	case avmNone:
+		return "none"
+	case avmAny:
+		return "any"
+	case avmUint64:
+		return "uint64"
+	case avmBytes:
+		return "[]byte"
+	}
+	return "internal error, unknown type"
+}
 
-	// StackBytes in an OpSpec shows that the op pops or yields a []byte
-	StackBytes
+var (
+	// StackUint64 is any valid uint64
+	StackUint64 = NewStackType(avmUint64, bound(0, math.MaxUint64))
+	// StackBytes is any valid bytestring
+	StackBytes = NewStackType(avmBytes, bound(0, maxStringSize))
+	// StackAny could be Bytes or Uint64
+	StackAny = StackType{
+		Name:        avmAny.String(),
+		AVMType:     avmAny,
+		ValueBound:  StackUint64.ValueBound,
+		LengthBound: StackBytes.LengthBound,
+	}
+	// StackNone is used when there is no input or output to
+	// an opcode
+	StackNone = StackType{
+		Name:    avmNone.String(),
+		AVMType: avmNone,
+	}
+
+	// StackBoolean constrains the int to 1 or 0, representing True or False
+	StackBoolean = NewStackType(avmUint64, bound(0, 1), "bool")
+	// StackHash represents output from a hash function or a field that returns a hash
+	StackHash = NewStackType(avmBytes, static(32), "hash")
+	// StackAddress represents a public key or address for an account
+	StackAddress = NewStackType(avmBytes, static(32), "addr")
+	// StackBigInt represents a bytestring that should be treated like an int
+	StackBigInt = NewStackType(avmBytes, bound(0, maxByteMathSize), "bigint")
+	// StackMethodSelector represents a bytestring that should be treated like a method selector
+	StackMethodSelector = NewStackType(avmBytes, static(4), "method")
+	// StackStorageKey represents a bytestring that can be used as a key to some storage (global/local/box)
+	StackStorageKey = NewStackType(avmBytes, bound(0, 64), "key")
```

Comment:
I guess here is where we could have:
`StackBoxKey = NewStackType(avmBytes, bound(1, 64), "bkey")`